api/v1/controllers/subjects.js

```diff
@@ -215,7 +216,25 @@ module.exports = {
    */
   findSubjects(req, res, next) {
     validateTags(null, req.swagger.params);
-    doFind(req, res, next, helper);
+    if (featureToggles.isFeatureEnabled(sampleStoreConstants.featureName) &&
+      featureToggles.isFeatureEnabled('getSubjectFromCache')) {
+      const resultObj = { reqStartTime: new Date() }; // for logging
+      redisSubjectModel.findSubjects(req, res, resultObj)
+      .then((response) => {
+        // loop through remove values to delete property
+        if (helper.fieldsToExclude) {
+          for (let i = response.length - 1; i >= 0; i--) {
+            u.removeFieldsFromResponse(helper.fieldsToExclude, response[i]);
+          }
+        }
+
+        u.logAPI(req, resultObj, response); // audit log
+        res.status(httpStatus.OK).json(response);
+      })
+      .catch((err) => u.handleError(next, err, helper.modelName));
+    } else {
+      doFind(req, res, next, helper);
+    }
   },
 
```

Comment:
Also, the populateSubjects gets only subjects that are published. So doFind may return different subjects. I talked to @iamigo about it and created another story to turn populateSubjects to return all subjects.